src/parsers/requests.js

```diff
@@ -62,10 +63,14 @@ export const getRequestDisplayDetails = (
     let message = payload?.params?.find(p => !isAddress(p));
     try {
       if (isHexString(message)) {
-        message = convertHexToUtf8(message);
+        message = toUtf8String(message);
```

Comment:
is this functionality different? if this is the root of the issue we should bring this up to walletconnect